src/dfs/ipfs.ts

```diff
@@ -248,4 +248,17 @@ export class Ipfs extends Logger implements DfsInterface {
   async remove(hash: string): Promise<any> {
     throw new Error('not implemented');
   }
+
+  /**
+   * Tries to decode given Buffer to UTF-8, if this leads to invalid characters, decode to Latin-1.
+   *
+   * @param      {Buffer}  buffer  buffer to decrypt, may be UTF-8 or Latin-1 encoded.
+   * @return     {string}  decoded string
+   */
+  private decodeBuffer(buffer: Buffer): string {
+    const decodedToUtf8 = buffer.toString('utf8');
+    return decodedToUtf8.indexOf('�') === -1
```

Comment:
seems a strange way to handle failed decoding, can't we use the `Buffer.isEncoding('utf8')` function beforehand?

Reply:
The Buffer.isEncoding() method returns true if the specified encoding is supported. So not matching the needed functionality :)

Reply:
At this point we have an already encdoded Buffer and try to decode it. `Buffer.isEncdoing` (despite its name) just checks if a certain encoding is supported:
```
Returns true if encoding contains a supported character encoding, or false otherwise
```
[docu](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_isencoding_encoding)